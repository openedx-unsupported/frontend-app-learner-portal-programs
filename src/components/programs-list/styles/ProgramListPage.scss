@import "@edx/brand/paragon/variables";
@import "@openedx/paragon/scss/core/core";
@import "@edx/brand/paragon/overrides";

// temporary override
div.menu button.menu-trigger {
  @extend .btn-outline-primary;
}

.title-override {
  font-family: 'Inter';
  font-weight: 700;
}
