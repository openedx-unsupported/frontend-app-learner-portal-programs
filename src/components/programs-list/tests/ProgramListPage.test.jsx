import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Alert } from '@edx/paragon';
import { IntlProvider } from 'react-intl';
import { getAuthenticatedUser } from '@edx/frontend-platform/auth';
import { ProgramListPage } from '../ProgramListPage';

jest.mock('@edx/frontend-platform/auth');
getAuthenticatedUser.mockReturnValue({
  username: 'edx',
});
jest.mock('@edx/frontend-platform/analytics');

describe('ProgramListPage', () => {
  const pageContext = {
    programs: [
      {
        programSlug: 'exampleprogram',
        programUUID: '6eefc008-db50-46f0-8746-667f55533a5d',
        programName: 'Example Program',
        programHostname: 'http://localhost:8734',
      },
      {
        programSlug: 'another-program',
        programUUID: '6eefc008-db50-46f0-8746-667f55533a5e',
        programName: 'Another Program',
        programHostname: 'http://localhost:8734/another-program',
      },
    ],
  };

  it('correctly renders the loading page', () => {
    const tree = renderer
      .create((
        <IntlProvider locale="en">
          <ProgramListPage
            isLoading
            pageContext={pageContext}
            fetchUserProgramEnrollments={jest.fn()}
          />
        </IntlProvider>
      ))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders fetching program error page when there are issues fetching the user programs', () => {
    const tree = renderer
      .create((
        <IntlProvider locale="en">
          <ProgramListPage
            isLoading={false}
            error={new Error()}
            pageContext={pageContext}
            fetchUserProgramEnrollments={jest.fn()}
          />
        </IntlProvider>
      ))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders table when there are valid programs', () => {
    const data = [
      {
        uuid: '6eefc008-db50-46f0-8746-667f55533a5d',
        name: 'Example Program',
        slug: 'exampleprogram',
      },
      {
        uuid: '6eefc008-db50-46f0-8746-667f55533a5e',
        name: 'Another Program',
        slug: 'another-program',
      },
    ];

    const wrapper = shallow((
      <ProgramListPage
        pageContext={pageContext}
        isLoading={false}
        fetchUserProgramEnrollments={jest.fn()}
      />
    ));

    wrapper.setProps({ enrolledPrograms: data });

    expect(wrapper.find('.table-responsive').exists()).toBeTruthy();
    expect(wrapper.find('tbody tr').length).toEqual(2);
  });

  it('renders error page when there are no valid programs', () => {
    const data = [
      {
        uuid: 'this-is-a-fake-program',
        name: 'This is a fake program',
        slug: 'fake-program',
      },
      {
        uuid: 'this-is-another-fake-program',
        name: 'This is another fake program',
        slug: 'another-fake-program',
      },
    ];

    const wrapper = shallow((
      <ProgramListPage
        pageContext={pageContext}
        isLoading={false}
        fetchUserProgramEnrollments={jest.fn()}
      />
    ));

    wrapper.setProps({ enrolledPrograms: data });

    expect(wrapper.find('.table-responsive').exists()).toBeFalsy();
    expect(wrapper.find(Alert).exists()).toBeTruthy();
  });
});
