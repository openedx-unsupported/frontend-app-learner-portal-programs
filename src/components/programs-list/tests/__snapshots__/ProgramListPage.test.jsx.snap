// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgramListPage correctly renders the loading page 1`] = `
<div>
  <main
    id="content"
  >
    <div
      className="container my-4"
    >
      <div
        className="loading-spinner d-flex justify-content-center align-items-center"
      >
        <div
          className="spinner-border text-primary"
          role="status"
        >
          <span
            className="sr-only"
          >
            loading program enrollments
          </span>
        </div>
      </div>
    </div>
  </main>
  <footer
    aria-label="edX Home"
    className="footer d-flex justify-content-center border-top py-3 px-4"
    role="contentinfo"
  >
    <div
      className="container-fluid d-grid"
    >
      <div
        className="area-1"
      >
        <a
          aria-label="edX Home"
          href="https://edx.org"
        >
          <img
            alt="edX Logo"
            style={
              Object {
                "maxHeight": 45,
              }
            }
          />
        </a>
      </div>
      <div
        className="area-2"
      >
        <h2>
          edX
        </h2>
        <ul
          className="list-unstyled p-0 m-0"
        >
          <li>
            <a
              href="https://edx.org/about-us"
            >
              About
            </a>
          </li>
          <li>
            <a
              href="https://edx.org/affiliate-program"
            >
              Affiliates
            </a>
          </li>
          <li>
            <a
              href="https://business.edx.org/?utm_campaign=edX.org+Referral&utm_medium=Footer&utm_source=localhost"
            >
              edX for Business
            </a>
          </li>
          <li>
            <a
              href="http://open.edx.org"
            >
              Open edX
            </a>
          </li>
          <li>
            <a
              href="https://edx.org/careers"
            >
              Careers
            </a>
          </li>
          <li>
            <a
              href="https://edx.org/news-announcements"
            >
              News
            </a>
          </li>
        </ul>
      </div>
      <div
        className="area-3"
      >
        <h2>
          Legal
        </h2>
        <ul
          className="list-unstyled p-0 m-0"
        >
          <li>
            <a
              href="https://edx.org/edx-terms-service"
            >
              Terms of Service & Honor Code
            </a>
          </li>
          <li>
            <a
              href="https://edx.org/edx-privacy-policy"
            >
              Privacy Policy
            </a>
          </li>
          <li>
            <a
              href="https://edx.org/accessibility"
            >
              Accessibility Policy
            </a>
          </li>
          <li>
            <a
              href="https://edx.org/trademarks"
            >
              Trademark Policy
            </a>
          </li>
          <li>
            <a
              href="https://edx.org/sitemap"
            >
              Sitemap
            </a>
          </li>
        </ul>
      </div>
      <div
        className="area-4"
      >
        <h2>
          Connect
        </h2>
        <ul
          className="list-unstyled p-0 m-0"
        >
          <li>
            <a
              href="https://blog.edx.org"
            >
              Blog
            </a>
          </li>
          <li>
            <a
              href="https://courses.edx.org/support/contact_us"
            >
              Contact Us
            </a>
          </li>
          <li>
            <a
              href="https://support.edx.org"
            >
              Help Center
            </a>
          </li>
          <li>
            <a
              href="https://edx.org/media-kit"
            >
              Media Kit
            </a>
          </li>
        </ul>
      </div>
      <div
        className="area-5"
      >
        <ul
          className="d-flex flex-row justify-content-between list-unstyled max-width-222 p-0 mb-4"
        >
          <li>
            <a
              href="http://www.facebook.com/EdxOnline"
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
              title="Facebook"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-facebook-square fa-w-14 fa-2x fa-rotate-null fa-pull-null social-icon"
                data-icon="facebook-square"
                data-prefix="fab"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M400 32H48A48 48 0 0 0 0 80v352a48 48 0 0 0 48 48h137.25V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.27c-30.81 0-40.42 19.12-40.42 38.73V256h68.78l-11 71.69h-57.78V480H400a48 48 0 0 0 48-48V80a48 48 0 0 0-48-48z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
              <span
                className="sr-only"
              >
                Like edX on Facebook
              </span>
            </a>
          </li>
          <li>
            <a
              href="https://twitter.com/edXOnline"
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
              title="Twitter"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-twitter-square fa-w-14 fa-2x fa-rotate-null fa-pull-null social-icon"
                data-icon="twitter-square"
                data-prefix="fab"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-48.9 158.8c.2 2.8.2 5.7.2 8.5 0 86.7-66 186.6-186.6 186.6-37.2 0-71.7-10.8-100.7-29.4 5.3.6 10.4.8 15.8.8 30.7 0 58.9-10.4 81.4-28-28.8-.6-53-19.5-61.3-45.5 10.1 1.5 19.2 1.5 29.6-1.2-30-6.1-52.5-32.5-52.5-64.4v-.8c8.7 4.9 18.9 7.9 29.6 8.3a65.447 65.447 0 0 1-29.2-54.6c0-12.2 3.2-23.4 8.9-33.1 32.3 39.8 80.8 65.8 135.2 68.6-9.3-44.5 24-80.6 64-80.6 18.9 0 35.9 7.9 47.9 20.7 14.8-2.8 29-8.3 41.6-15.8-4.9 15.2-15.2 28-28.8 36.1 13.2-1.4 26-5.1 37.8-10.2-8.9 13.1-20.1 24.7-32.9 34z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
              <span
                className="sr-only"
              >
                Follow edX on Twitter
              </span>
            </a>
          </li>
          <li>
            <a
              href="http://www.linkedin.com/company/edx"
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
              title="LinkedIn"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-linkedin fa-w-14 fa-2x fa-rotate-null fa-pull-null social-icon"
                data-icon="linkedin"
                data-prefix="fab"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
              <span
                className="sr-only"
              >
                Follow edX on LinkedIn
              </span>
            </a>
          </li>
          <li>
            <a
              href="https://www.instagram.com/edxonline/"
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
              title="Instagram"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-instagram fa-w-14 fa-2x fa-rotate-null fa-pull-null social-icon"
                data-icon="instagram"
                data-prefix="fab"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
              <span
                className="sr-only"
              >
                Follow edX on Instagram
              </span>
            </a>
          </li>
          <li>
            <a
              href="http://www.reddit.com/r/edx"
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
              title="Reddit"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-reddit-square fa-w-14 fa-2x fa-rotate-null fa-pull-null social-icon"
                data-icon="reddit-square"
                data-prefix="fab"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M283.2 345.5c2.7 2.7 2.7 6.8 0 9.2-24.5 24.5-93.8 24.6-118.4 0-2.7-2.4-2.7-6.5 0-9.2 2.4-2.4 6.5-2.4 8.9 0 18.7 19.2 81 19.6 100.5 0 2.4-2.3 6.6-2.3 9 0zm-91.3-53.8c0-14.9-11.9-26.8-26.5-26.8-14.9 0-26.8 11.9-26.8 26.8 0 14.6 11.9 26.5 26.8 26.5 14.6 0 26.5-11.9 26.5-26.5zm90.7-26.8c-14.6 0-26.5 11.9-26.5 26.8 0 14.6 11.9 26.5 26.5 26.5 14.9 0 26.8-11.9 26.8-26.5 0-14.9-11.9-26.8-26.8-26.8zM448 80v352c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V80c0-26.5 21.5-48 48-48h352c26.5 0 48 21.5 48 48zm-99.7 140.6c-10.1 0-19 4.2-25.6 10.7-24.1-16.7-56.5-27.4-92.5-28.6l18.7-84.2 59.5 13.4c0 14.6 11.9 26.5 26.5 26.5 14.9 0 26.8-12.2 26.8-26.8 0-14.6-11.9-26.8-26.8-26.8-10.4 0-19.3 6.2-23.8 14.9l-65.7-14.6c-3.3-.9-6.5 1.5-7.4 4.8l-20.5 92.8c-35.7 1.5-67.8 12.2-91.9 28.9-6.5-6.8-15.8-11-25.9-11-37.5 0-49.8 50.4-15.5 67.5-1.2 5.4-1.8 11-1.8 16.7 0 56.5 63.7 102.3 141.9 102.3 78.5 0 142.2-45.8 142.2-102.3 0-5.7-.6-11.6-2.1-17 33.6-17.2 21.2-67.2-16.1-67.2z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
              <span
                className="sr-only"
              >
                Subscribe to the edX subreddit
              </span>
            </a>
          </li>
        </ul>
        <ul
          className="d-flex flex-row justify-content-between list-unstyled max-width-264 p-0 mb-5"
        >
          <li>
            <a
              href="https://play.google.com/store/apps/details?id=org.edx.mobile"
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt="Download the edX mobile app from Google Play"
                className="max-height-39"
                src="https://prod-edxapp.edx-cdn.org/static/images/app/google_play_badge_45.6ea466e328da.png"
              />
            </a>
          </li>
          <li>
            <a
              href="https://itunes.apple.com/us/app/edx/id945480667?mt=8"
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt="Download the edX mobile app from the Apple App Store"
                className="max-height-39"
                src="https://prod-edxapp.edx-cdn.org/static/images/app/app_store_badge_135x40.d0558d910630.svg"
              />
            </a>
          </li>
        </ul>
        <p>
          © 
          2023
           edX LLC. All rights reserved.
          <br />
          <span>
            深圳市恒宇博科技有限公司 
            <a
              href="https://beian.miit.gov.cn"
              style={
                Object {
                  "textDecoration": "underline",
                }
              }
            >
              粤ICP备17044299号-2
            </a>
          </span>
        </p>
      </div>
    </div>
  </footer>
</div>
`;

exports[`ProgramListPage renders fetching program error page when there are issues fetching the user programs 1`] = `
<div>
  <main
    id="content"
  >
    <div
      className="container my-4"
    >
      <div
        className="fade alert-content alert alert-danger show"
        role="alert"
      >
        <span
          className="pgn__icon alert-icon"
        >
          <svg
            aria-hidden={true}
            fill="none"
            focusable={false}
            height={24}
            role="img"
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
              fill="currentColor"
            />
          </svg>
        </span>
        <div
          className="pgn__alert-message-wrapper"
        >
          <div
            className="alert-message-content"
          >
            An error occurred while fetching your program enrollments. Please try again later.
          </div>
        </div>
      </div>
    </div>
  </main>
  <footer
    aria-label="edX Home"
    className="footer d-flex justify-content-center border-top py-3 px-4"
    role="contentinfo"
  >
    <div
      className="container-fluid d-grid"
    >
      <div
        className="area-1"
      >
        <a
          aria-label="edX Home"
          href="https://edx.org"
        >
          <img
            alt="edX Logo"
            style={
              Object {
                "maxHeight": 45,
              }
            }
          />
        </a>
      </div>
      <div
        className="area-2"
      >
        <h2>
          edX
        </h2>
        <ul
          className="list-unstyled p-0 m-0"
        >
          <li>
            <a
              href="https://edx.org/about-us"
            >
              About
            </a>
          </li>
          <li>
            <a
              href="https://edx.org/affiliate-program"
            >
              Affiliates
            </a>
          </li>
          <li>
            <a
              href="https://business.edx.org/?utm_campaign=edX.org+Referral&utm_medium=Footer&utm_source=localhost"
            >
              edX for Business
            </a>
          </li>
          <li>
            <a
              href="http://open.edx.org"
            >
              Open edX
            </a>
          </li>
          <li>
            <a
              href="https://edx.org/careers"
            >
              Careers
            </a>
          </li>
          <li>
            <a
              href="https://edx.org/news-announcements"
            >
              News
            </a>
          </li>
        </ul>
      </div>
      <div
        className="area-3"
      >
        <h2>
          Legal
        </h2>
        <ul
          className="list-unstyled p-0 m-0"
        >
          <li>
            <a
              href="https://edx.org/edx-terms-service"
            >
              Terms of Service & Honor Code
            </a>
          </li>
          <li>
            <a
              href="https://edx.org/edx-privacy-policy"
            >
              Privacy Policy
            </a>
          </li>
          <li>
            <a
              href="https://edx.org/accessibility"
            >
              Accessibility Policy
            </a>
          </li>
          <li>
            <a
              href="https://edx.org/trademarks"
            >
              Trademark Policy
            </a>
          </li>
          <li>
            <a
              href="https://edx.org/sitemap"
            >
              Sitemap
            </a>
          </li>
        </ul>
      </div>
      <div
        className="area-4"
      >
        <h2>
          Connect
        </h2>
        <ul
          className="list-unstyled p-0 m-0"
        >
          <li>
            <a
              href="https://blog.edx.org"
            >
              Blog
            </a>
          </li>
          <li>
            <a
              href="https://courses.edx.org/support/contact_us"
            >
              Contact Us
            </a>
          </li>
          <li>
            <a
              href="https://support.edx.org"
            >
              Help Center
            </a>
          </li>
          <li>
            <a
              href="https://edx.org/media-kit"
            >
              Media Kit
            </a>
          </li>
        </ul>
      </div>
      <div
        className="area-5"
      >
        <ul
          className="d-flex flex-row justify-content-between list-unstyled max-width-222 p-0 mb-4"
        >
          <li>
            <a
              href="http://www.facebook.com/EdxOnline"
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
              title="Facebook"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-facebook-square fa-w-14 fa-2x fa-rotate-null fa-pull-null social-icon"
                data-icon="facebook-square"
                data-prefix="fab"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M400 32H48A48 48 0 0 0 0 80v352a48 48 0 0 0 48 48h137.25V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.27c-30.81 0-40.42 19.12-40.42 38.73V256h68.78l-11 71.69h-57.78V480H400a48 48 0 0 0 48-48V80a48 48 0 0 0-48-48z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
              <span
                className="sr-only"
              >
                Like edX on Facebook
              </span>
            </a>
          </li>
          <li>
            <a
              href="https://twitter.com/edXOnline"
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
              title="Twitter"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-twitter-square fa-w-14 fa-2x fa-rotate-null fa-pull-null social-icon"
                data-icon="twitter-square"
                data-prefix="fab"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-48.9 158.8c.2 2.8.2 5.7.2 8.5 0 86.7-66 186.6-186.6 186.6-37.2 0-71.7-10.8-100.7-29.4 5.3.6 10.4.8 15.8.8 30.7 0 58.9-10.4 81.4-28-28.8-.6-53-19.5-61.3-45.5 10.1 1.5 19.2 1.5 29.6-1.2-30-6.1-52.5-32.5-52.5-64.4v-.8c8.7 4.9 18.9 7.9 29.6 8.3a65.447 65.447 0 0 1-29.2-54.6c0-12.2 3.2-23.4 8.9-33.1 32.3 39.8 80.8 65.8 135.2 68.6-9.3-44.5 24-80.6 64-80.6 18.9 0 35.9 7.9 47.9 20.7 14.8-2.8 29-8.3 41.6-15.8-4.9 15.2-15.2 28-28.8 36.1 13.2-1.4 26-5.1 37.8-10.2-8.9 13.1-20.1 24.7-32.9 34z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
              <span
                className="sr-only"
              >
                Follow edX on Twitter
              </span>
            </a>
          </li>
          <li>
            <a
              href="http://www.linkedin.com/company/edx"
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
              title="LinkedIn"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-linkedin fa-w-14 fa-2x fa-rotate-null fa-pull-null social-icon"
                data-icon="linkedin"
                data-prefix="fab"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
              <span
                className="sr-only"
              >
                Follow edX on LinkedIn
              </span>
            </a>
          </li>
          <li>
            <a
              href="https://www.instagram.com/edxonline/"
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
              title="Instagram"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-instagram fa-w-14 fa-2x fa-rotate-null fa-pull-null social-icon"
                data-icon="instagram"
                data-prefix="fab"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
              <span
                className="sr-only"
              >
                Follow edX on Instagram
              </span>
            </a>
          </li>
          <li>
            <a
              href="http://www.reddit.com/r/edx"
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
              title="Reddit"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-reddit-square fa-w-14 fa-2x fa-rotate-null fa-pull-null social-icon"
                data-icon="reddit-square"
                data-prefix="fab"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M283.2 345.5c2.7 2.7 2.7 6.8 0 9.2-24.5 24.5-93.8 24.6-118.4 0-2.7-2.4-2.7-6.5 0-9.2 2.4-2.4 6.5-2.4 8.9 0 18.7 19.2 81 19.6 100.5 0 2.4-2.3 6.6-2.3 9 0zm-91.3-53.8c0-14.9-11.9-26.8-26.5-26.8-14.9 0-26.8 11.9-26.8 26.8 0 14.6 11.9 26.5 26.8 26.5 14.6 0 26.5-11.9 26.5-26.5zm90.7-26.8c-14.6 0-26.5 11.9-26.5 26.8 0 14.6 11.9 26.5 26.5 26.5 14.9 0 26.8-11.9 26.8-26.5 0-14.9-11.9-26.8-26.8-26.8zM448 80v352c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V80c0-26.5 21.5-48 48-48h352c26.5 0 48 21.5 48 48zm-99.7 140.6c-10.1 0-19 4.2-25.6 10.7-24.1-16.7-56.5-27.4-92.5-28.6l18.7-84.2 59.5 13.4c0 14.6 11.9 26.5 26.5 26.5 14.9 0 26.8-12.2 26.8-26.8 0-14.6-11.9-26.8-26.8-26.8-10.4 0-19.3 6.2-23.8 14.9l-65.7-14.6c-3.3-.9-6.5 1.5-7.4 4.8l-20.5 92.8c-35.7 1.5-67.8 12.2-91.9 28.9-6.5-6.8-15.8-11-25.9-11-37.5 0-49.8 50.4-15.5 67.5-1.2 5.4-1.8 11-1.8 16.7 0 56.5 63.7 102.3 141.9 102.3 78.5 0 142.2-45.8 142.2-102.3 0-5.7-.6-11.6-2.1-17 33.6-17.2 21.2-67.2-16.1-67.2z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
              <span
                className="sr-only"
              >
                Subscribe to the edX subreddit
              </span>
            </a>
          </li>
        </ul>
        <ul
          className="d-flex flex-row justify-content-between list-unstyled max-width-264 p-0 mb-5"
        >
          <li>
            <a
              href="https://play.google.com/store/apps/details?id=org.edx.mobile"
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt="Download the edX mobile app from Google Play"
                className="max-height-39"
                src="https://prod-edxapp.edx-cdn.org/static/images/app/google_play_badge_45.6ea466e328da.png"
              />
            </a>
          </li>
          <li>
            <a
              href="https://itunes.apple.com/us/app/edx/id945480667?mt=8"
              onClick={[Function]}
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt="Download the edX mobile app from the Apple App Store"
                className="max-height-39"
                src="https://prod-edxapp.edx-cdn.org/static/images/app/app_store_badge_135x40.d0558d910630.svg"
              />
            </a>
          </li>
        </ul>
        <p>
          © 
          2023
           edX LLC. All rights reserved.
          <br />
          <span>
            深圳市恒宇博科技有限公司 
            <a
              href="https://beian.miit.gov.cn"
              style={
                Object {
                  "textDecoration": "underline",
                }
              }
            >
              粤ICP备17044299号-2
            </a>
          </span>
        </p>
      </div>
    </div>
  </footer>
</div>
`;
